tests/FamilyHubs.ServiceDirectory.Api.FunctionalTests/BaseWhenUsingApiUnitTests.cs

```diff
@@ -11,25 +12,59 @@ public abstract class BaseWhenUsingApiUnitTests : IDisposable
     protected readonly HttpClient? Client;
     private readonly CustomWebApplicationFactory? _webAppFactory;
     private readonly bool _initSuccessful;
+    public static string BearerTokenSigningKey;
 
     protected BaseWhenUsingApiUnitTests()
     {
         try
         {
+            var configuration = new ConfigurationBuilder()
+                .AddUserSecrets<Program>()
+                .Build();
+
             _webAppFactory = new CustomWebApplicationFactory();
             _webAppFactory.SetupTestDatabaseAndSeedData();
 
             Client = _webAppFactory.CreateDefaultClient();
             Client.BaseAddress = new Uri("https://localhost:7128/");
 
             _initSuccessful = true;
+
+            BearerTokenSigningKey = IsRunningLocally(configuration)
+                ? configuration["GovUkOidcConfiguration:BearerTokenSigningKey"] : "StubPrivateKey123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ";
         }
         catch
         {
             _initSuccessful = false;
         }
     }
 
+    private bool IsRunningLocally(IConfiguration? configuration)
+    {
+        if (configuration == null)
+        {
+            return false;
+        }
+
+        try
+        {
+            string localMachineName = configuration["LocalSettings:MachineName"] ?? string.Empty;
+
+            if (!string.IsNullOrEmpty(localMachineName))
+            {
+                return Environment.MachineName.Equals(localMachineName, StringComparison.OrdinalIgnoreCase);
+            }
+        }
+        catch
+        {
+            return false;
+        }
+
+// Fallback to a default check if User Secrets file or machine name is not specified
+// For example, you can add additional checks or default behavior here
+        return false;
+    }
```

Comment:
This is a pretty big smell. I'd try and avoid anything that is explicitly trying to do something different during development.